File: typescript/easy/1281.subtractProductAndSum.ts
function subtractProductAndSum(n: number): number {
    let m = String(n)
        .split("")
        .reduce((acc, curr) => acc * Number(curr), 1)
    console.log(m)

    let s = String(n)
        .split("")
        .reduce((acc, curr) => acc + Number(curr), 0)

    return m - s
}

console.log(subtractProductAndSum(234)) // 15
// Output: 15
// Explanation:
// Product of digits = 2 * 3 * 4 = 24
// Sum of digits = 2 + 3 + 4 = 9
// Result = 24 - 9 = 15

File: python/easy/1281.subtractProductAndSum.py
class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        newList = []

        for x in str(n):
            newList.append(int(x))

        res_multiply = 1
        for y in newList:
            res_multiply = res_multiply * y

        res_sum = sum(newList)

        return res_multiply - res_sum


if __name__ == "__main__":
    solution = Solution()
    print(solution.subtractProductAndSum(234))  # 15

# Input: n = 234
# Output: 15
# Explanation:
# Product of digits = 2 * 3 * 4 = 24
# Sum of digits = 2 + 3 + 4 = 9
# Result = 24 - 9 = 15
